ui/src/app/shared/services/parser.spec.ts

```diff
@@ -1,508 +1,473 @@
+        expectChannels(node, 'a', 'b');
+    });
+
+    it('bridge error', () => {
+        parseResult = Parser.parse(':a > :b | bar', 'stream');
+        error = parseResult.lines[0].errors[0];
+        expect(error.message).toEqual('Expected app name but found \':\'');
+        expectRange(error.range, 5, 0, 6, 0);
+    });
+
+    it('named', () => {
+        parseResult = Parser.parse('aaaa=time | log', 'stream');
+        node = parseResult.lines[0].nodes[0];
+        expect(node.group).toEqual('aaaa');
+        expect(node.name).toEqual('time');
+        expect(node.type).toEqual('source');
+
+        node = parseResult.lines[0].nodes[1];
+        expect(node.group).toEqual('aaaa');
+        expect(node.name).toEqual('log');
+        expect(node.type).toEqual('sink');
+    });
+
+    it('multiline', () => {
+        parseResult = Parser.parse('time | log\nhdfs | file', 'stream');
+        node = parseResult.lines[0].nodes[0];
+        expect(node.name).toEqual('time');
+        node = parseResult.lines[0].nodes[1];
+        expect(node.name).toEqual('log');
+
+        node = parseResult.lines[1].nodes[0];
+        expect(node.name).toEqual('hdfs');
+        node = parseResult.lines[1].nodes[1];
+        expect(node.name).toEqual('file');
+        expectRange(node.range, 7, 1, 11, 1);
+    });
+
+    it('task', () => {
+        parseResult = Parser.parse('aaaa --bb=cc', 'task');
+        node = parseResult.lines[0].nodes[0];
+        expect(node.name).toEqual('aaaa');
+        expect(node.type).toEqual('task');
+        expect(node.options.get('bb')).toEqual('cc');
+        expectRange(node.optionsranges.get('bb'), 5, 0, 12, 0);
+    });
+
+    it('error: task with only name', () => {
+        parseResult = Parser.parse('aaaa=', 'task');
+        error = parseResult.lines[0].errors[0];
+        expect(error.message).toEqual('Expected format: name = taskapplication [options]');
+        expectRange(error.range, 0, 0, 0, 0);
+    });
+
+    it('error: task with extra data', () => {
+        parseResult = Parser.parse('aaaa=bbb ccc', 'task');
+        error = parseResult.lines[0].errors[0];
+        expect(error.message).toEqual('Unexpected data after task definition: ccc');
+        expectRange(error.range, 9, 0, 10, 0);
+    });
+
+    it('task with a name', () => {
+        parseResult = Parser.parse('bbb = aaaa --bb=cc', 'task');
+        expect(parseResult.lines.length).toEqual(1);
+        node = parseResult.lines[0].nodes[0];
+        expect(node.name).toEqual('aaaa');
+        expect(node.group).toEqual('bbb');
+        expect(node.options.get('bb')).toEqual('cc');
+    });
+
+    it('dotted arg names in stream', () => {
+        parseResult = Parser.parse('aaa --bbb.ccc.ddd=eee', 'stream');
+        expect(parseResult.lines.length).toEqual(1);
+        node = parseResult.lines[0].nodes[0];
+        expect(node.name).toEqual('aaa');
+        Array.from(node.options.keys()).forEach((k)=>console.log('key is '+k));
```

Comment:
Probably can remove this console.log.  (Will do on Merge).

Reply:
yep, good catch.